Yoast/wordpress-seo
# Route data via Redux for Advanced Tab

## Context


* We need Redux to be the source of truth for the Advanced tab, because there will be a period of time where the Advanced Settings are both visible in the metabox and in the modal. It is not yet clear how long this period will be.

## Summary


This PR can be summarized in the following changelog entry:

* Data source of truth for Advanced Tab is now our Redux store. Persistence still flows via hidden fields.

## Relevant technical choices:

* NOTE that the select2 fix we have implemented will be undone in a follow-up PR that is coming soon. We will switch to `react-select` soon, but decided to split that out to a new issue to keep the scope small. 

## Test instructions

This PR can be tested by following these steps:

* Test together with sibling PR on javascript: https://github.com/Yoast/javascript/pull/838
* Verify that the Advanced Tabs settings are correctly retrieved and are represented in our redux store (use browser plugin).
* Upon changing values, verify that the hidden fields and the store are reflecting those changes.
* Verify that advanced settings in the modal and in the metabox are both updated and in sync.
 

## UI changes

## Documentation

## Quality assurance


Fixes P1-142
